src/components/syntax-highlighter.js

```diff
@@ -0,0 +1,164 @@
+import { LitElement, html, css } from 'lit';
+import { unsafeHTML } from 'lit/directives/unsafe-html.js';
+import { marked } from 'marked';
+
+import Prism from 'prismjs';
+import './json-tree';
+
+// It's possible none of these imports are actually necessary and should just be removed
+import 'prismjs/components/prism-css';
+import 'prismjs/components/prism-yaml';
+import 'prismjs/components/prism-go';
+import 'prismjs/components/prism-ruby';
+import 'prismjs/components/prism-java';
+import 'prismjs/components/prism-json';
+import 'prismjs/components/prism-bash';
+import 'prismjs/components/prism-python';
+import 'prismjs/components/prism-http';
+import 'prismjs/components/prism-csharp';
+
+import PrismStyles from '../styles/prism-styles';
+import FontStyle from '../styles/font-styles';
+import InputStyle from '../styles/input-styles';
+import { getI18nText } from '../languages';
+import { copyToClipboard } from '../utils/common-utils';
+
+/**
+ * Mapping mime-type => prism language
+ */
+const LANGUAGES = [
+  {
+    pattern: /json/,
+    language: 'json'
+  }, {
+    pattern: /xml/,
+    language: 'html'
+  }
+];
+
+/**
+ * Syntax Highlighter component.
+ */
+class SyntaxHighlighter extends LitElement {
+  static get properties() {
+    return {
+      content: { type: Object },
+      language: { type: String, attribute: 'language' },
+      mimeType: { type: String, attribute: 'mime-type' },
+      copy: { type: Boolean, attribute: 'copy' }
```

Comment:
We use `copy` everywhere now, as far as I can tell, so this doesn't need to be passed in.